0018/README.md

```diff
@@ -0,0 +1,321 @@
+
+The following terms and definitions are used in this RFC.
+
+- **Event originator (originator)**: service that broadcasts endpoint filter events.
+- **Event listener (listener)**: service that subscribes to endpoint filter events.
+
+
+# NATS subject format
+
+Originators MUST publish endpoint filter events to the following NATS subject:
+```
+kaa.v1.events.{originator-service-instance-name}.endpoint-filter.{event-group}.{event-type}
+```
+
+where:
+- `{originator-service-instance-name}` is the name of the originator service instance. Allows listeners to subscribe to endpoint filter events from a specific originator.
+- `{event-group}` is a logical group of endpoint filter events. Equal to an endpoint filter identifier.
+- `{event-type}` is a specific endpoint filter event type.
+
+
+# Endpoint filter events
+
+
+## Endpoint filter activated
+
+Endpoint filter activated event is published when endpoint filter status changed from `pending` to `active`.
+The `{event-type}` is `activated`.
+
+Originators MUST use the following NATS subject format for endpoint filter activated events:
+```
+kaa.v1.events.{originator-service-instance-name}.endpoint-filter.{endpoint-filter-id}.activated
+```
+
+The NATS message payload is an Avro object with the following schema ([0018-epf-activated.avsc](./0018-epf-activated.avsc)):
+
+```json
+{
+    "namespace":"org.kaaproject.ipc.event.gen.v1.endpoint.filter",
+    "name":"EndpointFilterActivatedEvent",
+    "type":"record",
+    "doc":"Endpoint filter activated event message",
+    "fields":[
+        {
+            "name":"timestamp",
+            "type":"long",
+            "doc":"Message creation UNIX timestamp in milliseconds"
+        },
+        {
+            "name":"timeout",
+            "type":"long",
+            "default":0,
+            "doc":"Amount of milliseconds since the timestamp until the message expires. Value of 0 is reserved to indicate no expiration."
+        },
+        {
+            "name":"correlationId",
+            "type":"string",
+            "doc":"Message ID primarily used to track message processing across services"
+        },
+        {
+            "name":"originatorReplicaId",
+            "type":"string",
+            "doc":"Identifier of the service replica that generated the event"
+        },
+        {
+            "name":"id",
+            "type":"string",
+            "doc":"Identifier of the endpoint filter"
+        },
+        {
+            "name":"applicationName",
+            "type":"string",
+            "doc":"Application name the endpoint registered with"
+        },
+        {
+            "name":"name",
+            "type":[
+              "null",
+              "string"
+            ],
+            "doc":"Endpoint filter name, unique in scope of application"
+        }
+    ]
+}
+```
+
+
+## Endpoint filter deactivated(pending)
```

Comment:
Add space before the open parenthesis.

Reply:
Pending and deactivated should be different statuses. On what circumstances this event is published?